app/repositories/SessionRepository.scala

```diff
@@ -67,22 +70,28 @@ class DefaultSessionRepository @Inject() (config: AppConfig, component: Reactive
     )
   }
 
-  def get(key: String): Future[Option[Process]] = {
+  def get(key: String): Future[RequestOutcome[Process]] = {
     val updateModifier = Json.obj("$set" -> Json.obj("lastAccessed" -> Json.obj("$date" -> DateTime.now(DateTimeZone.UTC).getMillis)))
     findAndUpdate(Json.obj("_id" -> key), updateModifier, fetchNewObject = false)
-      .map(_.result[DefaultSessionRepository.SessionProcess].map(_.process))
+      .map(wr => wr.result[SessionProcess].fold(Left(DatabaseError): RequestOutcome[Process])(r => Right(r.process)))
       .recover {
         case lastError =>
           logger.error(s"Unable to retrieve process associated with key=$key")
-          None
+          Left(DatabaseError)
       }
   }
 
-  def set(key: String, process: Process): Future[Option[Unit]] =
-    insert(DefaultSessionRepository.SessionProcess(key, process.meta.id, process, DateTime.now(DateTimeZone.UTC))) map (_ => Some(())) recover {
-      case lastError =>
-        logger.error(s"Unable to persist process=${process.meta.id} against id=$key")
-        None
-    }
+  def set(key: String, process: Process): Future[RequestOutcome[Unit]] = {
+    val sessionDocument = Json.toJson(DefaultSessionRepository.SessionProcess(key, process.meta.id, process, DateTime.now(DateTimeZone.UTC)))
 
+    collection
```

Comment:
Following your recent observations should we be using "collection" here? Is there an appropriate method in ReactiveRepository?

Reply:
there doesnt seem to anything suitable on the main class and this one doesnt generate the deprecated warning

Reply:
Ok